flink-queryable-state/flink-queryable-state-client-java/src/main/java/org/apache/flink/queryablestate/network/AbstractServerBase.java

```diff
@@ -260,25 +262,47 @@ private boolean attemptToBind(final int port) throws Throwable {
 	/**
 	 * Shuts down the server and all related thread pools.
 	 */
-	public void shutdown() {
-		LOG.info("Shutting down server {} @ {}", serverName, serverAddress);
-
-		if (handler != null) {
-			handler.shutdown();
-			handler = null;
-		}
-
-		if (queryExecutor != null) {
-			queryExecutor.shutdown();
-		}
+	public CompletableFuture<?> shutdownServer(Time timeout) throws InterruptedException {
```

Comment:
Ideally we would get rid of the `timeout` parameter because that is something the user specifies when calling `get()` on the returned termination future.

Reply:
I agree, but we need the `timeout` in order to shut down the `executors`.